### PR TITLE
Fix react native example and bootstrapping

### PR DESCRIPTION
**Pull request by shilman**
Issue: #1509 

## What I did

Addresses issues raised in #1509:
- bootstrapping react native doesn't work
- error-logging channel breaks RN, so reverting this (cc @tmeasday)

## How to test
```
yarn bootstrap && yarn bootstrap:react-native-vanillla
cd examples/react-native-vanilla
yarn storybook
./node_modules/.bin/react-native run-ios
```

### CODE REVIEW

File: lib/addons/src/index.js [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
-}
-
 export class AddonStore {
   constructor() {
     this.loaders = {};
     this.panels = {};
     // this.channel should get overwritten by setChannel if package versions are
     // correct and AddonStore is a proper singleton. If not, throw an error.
-    this.channel = { on: channelError, emit: channelError };
+    this.channel = null;
```
Comment (ndelangen) [inline lib/addons/src/index.js]: it was good whilst it lasted

**Review by ndelangen — APPROVED** [review]

File: lib/addons/src/index.js (hunk 1 above)
Comment (shilman) [inline lib/addons/src/index.js]: It wasn't good! It wasted multiple hours of my time! 😭

**Review by tmeasday — APPROVED** [review]
Not sure about the bootstrapping bit, but good with removing the check.

**Comment by tmeasday** [comment]
An alternative approach: https://github.com/storybooks/storybook/pull/1515

**Pull request merged**